### PR TITLE
Experius patch registration address

### PR DESCRIPTION
**Pull request by lewisvoncken**
### Description

Add configuration setting to Show Address Fields on the Registration page

### Manual testing scenarios

1. View the Customer Account Create
2. Apply changes
3. Enable the configuration setting (Stores > Settings > Configuration > Customer > Customer Configuration > Create New Account Options > Show Address Fields
4. View the Customer Account Create and fill in the form

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/Customer/Block/Form/Register.php [hunk 1]
```diff
@@ -220,4 +244,35 @@ public function getRequiredCharacterClassesNumber()
     {
         return $this->_scopeConfig->getValue(AccountManagement::XML_PATH_REQUIRED_CHARACTER_CLASSES_NUMBER);
     }
+
+    /**
+     * Determine if Address Fields have to be shown
+     *
+     * @return bool
+     */
+    public function getShowAddressFields()
```
Comment (vrann) [inline app/code/Magento/Customer/Block/Form/Register.php]: where is this method used?

File: app/code/Magento/Customer/Block/Form/Register.php [hunk 2]
```diff
@@ -220,4 +244,35 @@ public function getRequiredCharacterClassesNumber()
...
+     * @param string $formCode
+     * @return array
+     */
+    public function isAddressFieldUsedInForm($attributeCode)
+    {
+        $formCode = 'customer_register_address';
+        $entityType = AddressMetadataInterface::ENTITY_TYPE_ADDRESS;
+        $attributeId = $this->_eavAttribute->getIdByCode( $entityType, $attributeCode);
+        $bind = ['attribute_id' => $attributeId, 'form_code' => $formCode];
+        $select = $this->_resource->getConnection()->select()->from(
```
Comment (vrann) [inline app/code/Magento/Customer/Block/Form/Register.php]: It is not a recommended approach having SQL queries in the Blocks. Ideally they should be placed in resource models.

**Review by vrann — CHANGES_REQUESTED** [review]

File: app/code/Magento/Customer/Block/Form/Register.php (hunk 1 above)
Comment (lewisvoncken) [reply app/code/Magento/Customer/Block/Form/Register.php]: @vrann this method is used in the template it was already there but not in use

**Comment by lewisvoncken** [comment]
@vrann I have just made changes based on the feedback

**Comment by lewisvoncken** [comment]
@vrann do you have any questions ?

**Comment by magento-engcom-team** [comment]
@lewisvoncken thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Review by okorshenko — APPROVED** [review]

**Comment by okorshenko** [comment]
Hi @lewisvoncken 
Here is what we got from QA department:
QA/QC Report
## Steps to reproduce

1. Enable the configuration setting (Stores > Settings > Configuration > Customer > Customer Configuration > Create New Account Options > Show Address Fields
2. Flush cache
2. Got to store front. Customer Account Create page and fill in the form

✅ Expected Result: Text field State/Province present.

 ❌ Actual result: Not present text field State/Province
stateprovince


 Note: I can create customer with blank required State/Province field.

**Comment by okorshenko** [comment]
Hi @lewisvoncken 
You can see this behavior here:
http://pr-11655.engcom.dev.magento.com/
(Admin access: admin: 123123q)

**Comment by okorshenko** [comment]
Hi @lewisvoncken please let us know if you would like proceed with this PR

**Comment by okorshenko** [comment]
Hi @lewisvoncken  We are closing this PR due to inactivity. Feel free to reopen it once ready

**Pull request closed without merging**